src/acme/ui/CustomerCreatePanel.java

```diff
@@ -0,0 +1,144 @@
+import javax.swing.JLabel;
+import java.awt.Font;
+import javax.swing.JTextField;
+import javax.swing.JComboBox;
+import javax.swing.JButton;
+
+public class CustomerCreatePanel extends AcmeBaseJPanel {
+	Company company;
+	Customer customer;
+	private JTextField nameField;
+	private JLabel numberField;
+	private JComboBox<Road> avenueComboField;
+	private JComboBox<Road> streetComboField;
+
+	public CustomerCreatePanel() {
+        super();
+        customer = new Customer();
+        customer.create();
+    }
+	
+	@Override
+    public void buildPanel() {
+        company = this.getCompany();
+        this.removeAll();
+        
+        setLayout(null);
+        
+        JLabel titleLabel = new JLabel("Customer");
+        titleLabel.setBounds(42, 30, 75, 22);
+        titleLabel.setFont(new Font("Tahoma", Font.PLAIN, 18));
+        add(titleLabel);
+        
+        JLabel nameLabel = new JLabel("Name");
+        nameLabel.setBounds(57, 88, 46, 14);
+        add(nameLabel);
+        
+        nameField = new JTextField();
+        nameField.setBounds(113, 85, 125, 20);
+        add(nameField);
+        nameField.setColumns(10);
+        
+        JLabel numberLabel = new JLabel("Number");
+        numberLabel.setBounds(57, 126, 46, 14);
+        add(numberLabel);
+        
+        
+        numberField = new JLabel(Integer.toString(Customer.getNextCustomerNumber()));
+        numberField.setBounds(113, 126, 46, 14);
+        add(numberField);
+        
+        JLabel lblIntersection = new JLabel("Intersection");
+        lblIntersection.setFont(new Font("Tahoma", Font.PLAIN, 14));
+        lblIntersection.setBounds(57, 166, 75, 22);
+        add(lblIntersection);
+        
+        JLabel avenueLabel = new JLabel("Avenue");
+        avenueLabel.setBounds(57, 199, 46, 14);
+        add(avenueLabel);
+        
+        JLabel streetLabel = new JLabel("Street");
+        streetLabel.setBounds(57, 240, 46, 14);
+        add(streetLabel);
+        
+        Set<Road> avenues = company.getMap().getAvenues();
+        avenueComboField = new JComboBox<Road>(avenues.toArray(new Road[0]));
+        avenueComboField.setBounds(113, 196, 125, 20);
+        add(avenueComboField);
+        
+        Set<Road> streets = company.getMap().getStreets();
+        streetComboField = new JComboBox<Road>(streets.toArray(new Road[0]));
+        streetComboField.setBounds(113, 237, 125, 20);
+        add(streetComboField);
+        
+        JButton saveButton = new JButton("Save");
+        saveButton.addActionListener(e -> saveCustomer());
+        saveButton.setBounds(330, 266, 89, 23);
+        add(saveButton);
+        
+        JButton cancelButton = new JButton("Cancel");
+        cancelButton.addActionListener(e -> cancelCreate());
+        cancelButton.setBounds(231, 266, 89, 23);
+        add(cancelButton);
+        
+        this.revalidate();
+        this.repaint();
+	}
+	
+	private void navigateToPreviousPage() {
+		this.getAcmeUI().courierList();

```

Comment:
This should be customerList()

Reply:
And what if you are accessing the new customer page from the Ticket Creation menu?

Reply:
Got it. `CustomerCreatePanel` now handles context-based `onSave` navigation for Customer List vs Ticket Creation.